logbook-core/src/main/java/org/zalando/logbook/DefaultLogbookFactory.java

```diff
@@ -42,15 +40,15 @@ public Logbook create(
             @Nullable final HttpLogWriter writer) {
 
 
-        final HeaderObfuscator header = firstNonNull(headerObfuscator, authorization());
-        final BodyObfuscator body = firstNonNull(bodyObfuscator, BodyObfuscator.none());
+        final HeaderObfuscator header = Optional.ofNullable(headerObfuscator).orElseGet(Obfuscators::authorization);
+        final BodyObfuscator body = Optional.ofNullable(bodyObfuscator).orElseGet(BodyObfuscator::none);
 
         return new DefaultLogbook(
-                firstNonNull(condition, $ -> true),
+                Optional.ofNullable(condition).orElse($ -> true),
                 combine(queryObfuscator, header, body, requestObfuscator),
                 combine(header, body, responseObfuscator),
-                firstNonNull(formatter, new DefaultHttpLogFormatter()),
-                firstNonNull(writer, new DefaultHttpLogWriter())
+                Optional.ofNullable(formatter).orElseGet(DefaultHttpLogFormatter::new),
```

Comment:
Nice, looks like previously we always unnecessarily created the default instances.